pkg/asset/manifests/tectonic.go

```diff
@@ -39,8 +41,45 @@ func (t *Tectonic) Generate(dependencies asset.Parents) error {
 	ingressCertKey := &tls.IngressCertKey{}
 	kubeCA := &tls.KubeCA{}
 	dependencies.Get(installConfig, ingressCertKey, kubeCA)
+	var awsCreds AWS
+	var libvirtCreds Libvirt
+	var openStackCreds OpenStack
+	var cloudCredsData []byte
+	switch {
+	case installConfig.Config.Platform.AWS != nil:
+		p := credentials.SharedCredentialsProvider{}
+		awscreds, err := p.Retrieve()
+		if err != nil {
+			return err
+		}
+		awsCreds = AWS{
```

Comment:
Instead of using temporary `AWS`, `Libvirt`, and `OpenStack` variables, create `cloudCreds` here.

```
cloudCreds := cloudCredsTemplateData{
    AWS: AWS{
        Base64encodeAccessKeyID:     base64.StdEncoding.EncodeToString([]byte(awscreds.AccessKeyID)),
        Base64encodeSecretAccessKey: base64.StdEncoding.EncodeToString([]byte(awscreds.SecretAccessKey)),
    },
}
```

Reply:
:+1:

Reply:
I still have to initialize that cloudCreds var in order to use it below, unless i'm missing something.  I got rid of the temp AWS/Libvirt/OpenStack vars